### PR TITLE
#167995472 Register a customer

### PR DESCRIPTION
**Pull request by shaolinmkz**
What does this PR do?
Register a user

Description of Task to be completed?
- Build out REST API to signup users
- User receives a confirmation email on signup
- Write test

How should this be manually tested?
Clone this branch.
Run cd into shoplink-backend
Run npm install to install the dependencies.
Run npm test to run the tests.

Any background context you want to provide?

What are the relevant pivotal tracker stories?
[#167995472](https://www.pivotaltracker.com/story/show/167995472)

### CODE REVIEW

**Comment by kleva-j** [comment]
I think we should get rid of the Hound stuff. These errors all over the place are not making sense.

**Comment by kleva-j** [comment]
I think we should consider file linking/importing using `absolute` path.

File: server/middleware/validateSignUpFields.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ * @returns {object} returns an object
+ */
+const checkTemplate = (field) => check(field)
+  .exists({
+    checkNull: true,
+    checkFalsy: true
+  });
+
+export default [
+  checkTemplate('firstName')
```
[the remark this answers is not in the document]
Comment (shaolinmkz) [reply server/middleware/validateSignUpFields.js]: This code was already refactored due to the initial complaint made

File: server/controller/index.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import Users from './Users';
+
+const removeMe = { remove: 'me' };
```
Comment (kleva-j) [inline server/controller/index.js:3]: Why do you have this variable here.

**Review by kleva-j — APPROVED** [review]

**Pull request merged**